vertices.go

```diff
@@ -70,85 +97,25 @@ func vertices(sx0, sy0, sx1, sy1 int, width, height int, geo *affine.GeoM) []flo
 		g.Concat(geo)
 		geo = &g
 	}
+	x0, y0 := float64(0), float64(0)
+	x1, y1 := float64(sx1 - sx0), float64(sy1 - sy0)
 
-	a, b, c, d, tx, ty := geo.Elements()
-	g0 := float32(a)
-	g1 := float32(b)
-	g2 := float32(c)
-	g3 := float32(d)
-	g4 := float32(tx)
-	g5 := float32(ty)
-	w := 1
-	h := 1
-	for w < width {
-		w *= 2
-	}
-	for h < height {
-		h *= 2
-	}
+	// it really feels like we should be able to cache this computation
+	// but it may not matter.
+	w := npo2(width)
+	h := npo2(height)
 	wf := float32(w)
 	hf := float32(h)
-	x0, y0, x1, y1 := float32(0), float32(0), float32(sx1-sx0), float32(sy1-sy0)
 	u0, v0, u1, v1 := float32(sx0)/wf, float32(sy0)/hf, float32(sx1)/wf, float32(sy1)/hf
 
-	// Vertex coordinates
-	vs[0] = x0
-	vs[1] = y0
-
-	// Texture coordinates: first 2 values indicates the actual coodinate, and
-	// the second indicates diagonally opposite coodinates.
-	// The second is needed to calculate source rectangle size in shader programs.
-	vs[2] = u0
-	vs[3] = v0
-	vs[4] = u1
-	vs[5] = v1
-
-	// Geometry matrix
-	vs[6] = g0
-	vs[7] = g1
-	vs[8] = g2
-	vs[9] = g3
-	vs[10] = g4
-	vs[11] = g5
-
-	vs[12] = x1
-	vs[13] = y0
-	vs[14] = u1
-	vs[15] = v0
-	vs[16] = u0
-	vs[17] = v1
-	vs[18] = g0
-	vs[19] = g1
-	vs[20] = g2
-	vs[21] = g3
-	vs[22] = g4
-	vs[23] = g5
-
-	vs[24] = x0
-	vs[25] = y1
-	vs[26] = u0
-	vs[27] = v1
-	vs[28] = u1
-	vs[29] = v0
-	vs[30] = g0
-	vs[31] = g1
-	vs[32] = g2
-	vs[33] = g3
-	vs[34] = g4
-	vs[35] = g5
-
-	vs[36] = x1
-	vs[37] = y1
-	vs[38] = u1
-	vs[39] = v1
-	vs[40] = u0
-	vs[41] = v0
-	vs[42] = g0
-	vs[43] = g1
-	vs[44] = g2
-	vs[45] = g3
-	vs[46] = g4
-	vs[47] = g5
+	x, y := geo.Apply32(x0, y0)
+	oneVertex(vs[0:6], x, y, u0, v0, u1, v1)
+	x, y = geo.Apply32(x1, y0)
+	oneVertex(vs[6:12], x, y, u1, v0, u0, v1)
+	x, y = geo.Apply32(x0, y1)
+	oneVertex(vs[12:18], x, y, u0, v1, u1, v0)
+	x, y = geo.Apply32(x1, y1)
+	oneVertex(vs[18:24], x, y, u1, v1, u0, v0)
```

Comment:
These create new slices: Is the performance on browsers fine?

Reply:
I haven't tried, I'm not sure how. It may be worse, in which case probably don't do this?

Reply:
As we discussed on Slack, this would affect the performance on browsers since a slice is represented by ArrayBuffer. Please keep these assignments flat.